### PR TITLE
docs: restore non-depends build documentation

### PR DESCRIPTION
**Pull request by kittywhiskers**
### Purpose

After https://github.com/dashpay/dash/pull/5077, Dash is no longer tied to the its `depends` and can be built from distro-obtained dependencies as it was before. Work on re-aligning Dash's build system with Bitcoin's has been documented in https://github.com/dashpay/dash-issues/issues/51. This PR aims to realign build documentation with upstream.

### Notes

* `revert: revert dash#2323 (Update build documentation)` is not a _true_ revert as after dash#2323, backports were done ([source](https://github.com/dashpay/dash/commits/b026f86903136441dbbfa286d863aaaa6945abe9/doc/build-unix.md)), including but not limited to, documentation updates for OpenSSL removal, SQLite & `libnatpmp` introduction and formatting changes. 

   That revert commit attempts to preserve the spirit of the backports done after it while reverting the `depends`-centric documentation changes but as a result, changes may be misattributed and so closer review is requested.

* `bison` is a dependency needed for building depends (as visible [here](https://github.com/dashpay/dash/blame/b026f86903136441dbbfa286d863aaaa6945abe9/depends/README.md#L59)) but isn't required to build `dash-qt` if building from dependencies sourced from your distro's package manager.

    Despite that, `bison` has been retained as a dependency for now based on feedback.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -48,6 +48,7 @@ OSX_QT_TRANSLATIONS = ar,bg,ca,cs,da,de,es,fa,fi,fr,gd,gl,he,hu,it,ja,ko,lt,lv,p
 
 DIST_CONTRIB = \
            $(top_srcdir)/contrib/debian/copyright \
+		   $(top_srcdir)/contrib/install_db4.sh \
```
Comment (UdjinM6) [inline Makefile.am:51]: 11702: nit: whitespaces

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
👍 

linter complains https://gitlab.com/dashpay/dash/-/jobs/4166362049 - `contrib/install_db4.sh` needs more work

File: Makefile.am (hunk 1 above)
Comment (kittywhiskers) [reply Makefile.am]: Backported ShellCheck fixes in latest push!

**Comment by PastaPastaPasta** [comment]
Linter fails still

File: doc/build-osx.md [hunk 2]
```diff
@@ -17,29 +17,61 @@ When the popup appears, click `Install`.
 
 Then install [Homebrew](https://brew.sh).
 
-## Base build dependencies
-
+## Dependencies
 ```shell
-brew install automake libtool pkg-config libnatpmp sqlite
+brew install automake berkeley-db4 libtool boost miniupnpc pkg-config python qt libevent libnatpmp qrencode sqlite
```
Comment (PastaPastaPasta) [inline doc/build-osx.md:22]: we need gmp as well. and even with it installed ./configure doesn't find it

Comment (kittywhiskers) [reply doc/build-osx.md:22]: Then should we include it or add it once detection has been fixed?

**Comment by PastaPastaPasta** [comment]
I lean towards saying we should make sure these docs work before merging in :)

**Comment by PastaPastaPasta** [comment]
rebasing to pull in 5394

File: doc/build-osx.md (hunk 2 above)
Comment (PastaPastaPasta) [reply doc/build-osx.md:22]: Okay, we should add gmp stuff here now

Comment (kittywhiskers) [reply doc/build-osx.md]: Documentation has been updated for macOS and Linux (Fedora and Debian flavours)

**Review by UdjinM6 — APPROVED** [review]
LGTM,  utACK

**Review by PastaPastaPasta — COMMENTED** [review]
post-utACK merge

**Pull request merged**